pkg/errors/gatherer.go

```diff
@@ -44,6 +47,32 @@ func (g *Gatherer) Failed() <-chan struct{} {
 	return g.failed
 }
 
+// DoneOrFailed closes the returned channel when either:
+// all routines returned successfully OR
+// any routine returned with an error.
+func (g *Gatherer) DoneOrFailed() <-chan struct{} {
+	return g.DoneOrFailedCtx(context.Background())
+}
+
+// DoneOrFailedCtx closes the returned channel when either:
+// all routines returned successfully OR
+// any routine returned with an error OR
+// the context was cancelled.
+func (g *Gatherer) DoneOrFailedCtx(ctx context.Context) <-chan struct{} {
```

Comment:
Make it return `bool`, indicating whether it was done or failed within the context deadline. Remove the goroutine. Rename to `WaitDoneOrFailedCtx()`.

Reply:
done